src/main/java/htsjdk/samtools/cram/CRAIEntry.java

```diff
@@ -73,30 +82,18 @@ public void writeToStream(OutputStream os) {
      * Format the entry as a string suitable for serialization in the CRAI index
      */
     private String serializeToString() {
-        return String.format(entryFormat,
+        return String.format(ENTRY_FORMAT,
                 sequenceId, alignmentStart, alignmentSpan,
-                containerStartOffset, sliceOffset, sliceSize);
+                containerStartOffset, sliceByteOffset, sliceByteSize);
     }
 
     @Override
     public String toString() { return serializeToString(); }
 
     public static List<CRAIEntry> fromContainer(final Container container) {
-        final List<CRAIEntry> entries = new ArrayList<>(container.slices.length);
-        for (int i = 0; i < container.slices.length; i++) {
-            final Slice s = container.slices[i];
-            final CRAIEntry e = new CRAIEntry();
-            e.sequenceId = s.sequenceId;
-            e.alignmentStart = s.alignmentStart;
-            e.alignmentSpan = s.alignmentSpan;
-            e.containerStartOffset = s.containerOffset;
-            e.sliceOffset = container.landmarks[i];
-            e.sliceSize = s.size;
-
-            e.sliceIndex = i;
-            entries.add(e);
-        }
-        return entries;
+        return Arrays.stream(container.slices)
+                .map(slice -> slice.getCRAIEntry(slice.containerOffset))
```

Comment:
Can't necessarily derive the container offset from the Slice because sometimes it conflict's with the Container's own offset - will need to look into why.

Reply:
Also, any reason not to move this method to container `getCRAIEntries` just as it is, and have  it return the list of entries. Eventually it should end up there I think, and would be similar to the new one you added to `Slice`.

Reply:
OK, these suggestions make sense.

Slice.containerOffset is the byte offset of the Container from the beginning of the stream.

Slice.offset is the byte offset of the Slice from the beginning of its Container.

I'll add comments making this more clear to `Slice`